pkg/hwevent/event_data.go

```diff
@@ -16,17 +16,197 @@ package hwevent
+	}
+	b.WriteString("	MessageId: " + e.MessageID + "\n")
+	if e.Oem != nil {
+		b.WriteString("	Oem: " + string(e.Oem) + "\n")
+	}
+	if e.OriginOfCondition != "" {
+		b.WriteString("	OriginOfCondition: " + e.OriginOfCondition + "\n")
+	}
+	if e.Severity != "" {
+		b.WriteString("	Severity: " + e.Severity + "\n")
+	}
+	if e.Resolution != "" {
+		b.WriteString("	Resolution: " + e.Resolution + "\n")
+	}
+	return b.String()
+}
+
+// RedfishEvent is defined in Redfish schema Event.v1_4_1.json
+// https://redfish.dmtf.org/schemas/v1/Event.v1_4_1.json
+// The Event schema describes the JSON payload received by an Event Destination,
+// which has subscribed to event notification, when events occur.  This Resource
+// contains data about events, including descriptions, severity, and a MessageId
+// link to a Message Registry that can be accessed for further information.
+//
+// Required fields: @odata.type, Events, Id, Name
+type RedfishEvent struct {
+	// +optional
+	OdataContext string `json:"@odata.context,omitempty"`
+	// +required
+	OdataType string `json:"@odata.type"`
+	// The available actions for this resource.
+	// +optional
+	Actions []byte `json:"Actions,omitempty"`
+	// A context can be supplied at subscription time.  This property
+	// is the context value supplied by the subscriber.
+	// +optional
+	Context string `json:"Context,omitempty"`
+	// +optional
+	Description string `json:"Description,omitempty"`
+	// +required
+	Events []EventRecord `json:"Events"`
+	// +required
+	ID string `json:"Id"`
+	// +required
+	Name string `json:"Name"`
+	// This is the manufacturer/provider specific extension
+	// +optional
+	Oem []byte `json:"Oem,omitempty"`
+}
+
+// String returns a pretty-printed representation of the RedfishEvent.
+func (e RedfishEvent) String() string {
+	b := strings.Builder{}
+	if e.OdataContext != "" {
+		b.WriteString("\n@odata.context: " + e.OdataContext + "\n")
+	}
+	b.WriteString("@odata.type: " + e.OdataType + "\n")
+	if e.Actions != nil {
+		b.WriteString("Actions: " + string(e.Actions) + "\n")
+	}
+	if e.Context != "" {
+		b.WriteString("Context: " + e.Context + "\n")
+	}
+	b.WriteString("Id: " + e.ID + "\n")
+	b.WriteString("Name: " + e.Name + "\n")
+	if e.Oem != nil {
+		b.WriteString("Oem: " + string(e.Oem) + "\n")
+	}
+	for i, e := range e.Events {
+		b.WriteString("Events[" + strconv.Itoa(i) + "]:\n")
+		b.WriteString(e.String())
+	}
+	return b.String()
+}
+
 // Data ... cloud native events data
 // Data Json payload is as follows,
 //{
-//	"version": "v1.0",
+//	"version": "v1.4.1",
 //
 //}
 type Data struct {
-	Version string `json:"version" example:"v1"`
-	Data    []byte `json:"data"`
+	Version string        `json:"version" example:"v1"`
+	Data    *RedfishEvent `json:"data"`
+}
+
+// String returns a pretty-printed representation of the Data.
+func (d Data) String() string {
+	b := strings.Builder{}
+	b.WriteString("version: " + d.Version + "\n")
+	b.WriteString("data:\n")
+	b.WriteString(d.Data.String())
```

Comment:
No need for a new line after data, d.data string function should prettify it too